buildSrc/build.gradle.kts

```diff
@@ -11,4 +11,6 @@ val kotlinVersion = "1.9.22"
 
 dependencies {
     implementation("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion")
+    implementation("io.spring.gradle:dependency-management-plugin:1.1.4")
+    implementation("org.springframework.boot:spring-boot-gradle-plugin:3.2.4")
```

Comment:
to use external plugins like Spring we have to add this plugin to root build src with our target version
https://docs.gradle.org/current/userguide/custom_plugins.html#applying_external_plugins_in_precompiled_script_plugins